metals/src/main/scala/scala/meta/internal/metals/Tables.scala

```diff
@@ -161,6 +168,47 @@ final class Tables(
     }
   }
 
+  /**
+   * Between h2 "2.1.x" and "2.2.x" write/read formats in MVStore changed
+   * (https://github.com/h2database/h2database/pull/3834)
+   */
+  private def upgradeIfNeeded(url: String): Unit = {
+    val oldVersion = 214
+    val formatVersionChangedMessage =
+      "The write format 2 is smaller than the supported format 3"
```

Comment:
Any chance we can match on an exact error?

Reply:
We could not check the exact message and just the error code. Trouble is that we wouldn't know the versions  are`2` and `3`, so if it changes again in the future, it might be `3` and `4`. Maybe it doesn't matter so much since it will fail if the versions change again anyway. Otherwise (looking at the source code), the versions in the error are only in the message I'm afraid.